Add phone calls & available models (#26)
fixie-ai/ultradox: docs/src/content/docs/guides/phonecalls.mdx

---
title: "Phone Calls"
description: Use Ultravox API to make and receive calls from regular phones via Twilio.
---

import { Steps, Tabs, TabItem } from '@astrojs/starlight/components';

## Overview

The Ultravox API allows you to create AI-powered voice applications that can interact with regular phone numbers. This enables Ultravox AI agents to make outgoing calls and receive incoming calls from traditional phone networks.

:::note[Twilio Support]
We currently integrate with Twilio. Please let us know if there's another integration you'd like to see.
:::

This guide will walk you through the process of setting up and using the Ultravox API with Twilio for both outgoing and incoming phone calls.

## Creating a Phone Call with Twilio
:::tip[Prerequisites]
Make sure you have:
1. An active Twilio account
1. A phone number purchased from Twilio
:::

> Creating an Ultravox call that works with Twilio is just like creating a WebRTC call except you must add two additional parameters to the [Create Call](./api/calls/#create-call) command:

Reviewer: ```suggestion
Creating an Ultravox call that works with Twilio is just like creating a WebRTC call, but there are two parameters to the [Create Call](./api/calls/#create-call) command worth special attention:
```

<table class="w-full">
    <tr class="w-full">
        <th class="w-1/12"></th>
        <th class="w-1/12"></th>
        <th class="w-10/12"></th>
    </tr>
    <tr>
        <td class="font-mono">medium</td>
        <td>object</td>
        <td>Tells Ultravox which protocol to use. <br />For Twilio, must be set to `{"twilio": {}}` and sets the call to use Twilio [Media Streams](https://www.twilio.com/docs/voice/media-streams). Defaults to `{"webRtc": {}}` which sets the protocol to WebRTC.</td>
    </tr>
    <tr>
        <td class="font-mono">initiator</td>
        <td>string</td>

>         <td>Tells Ultravox who is responsible for starting the call. For outgoing calls, typically set to `"agent"`. Default is `"user"`.</td>

Reviewer: ```suggestion
        <td>Tells Ultravox who started the call. For outgoing calls, typically set to `"INITIATOR_AGENT"`. Default is `"INITIATOR_USER"`.</td>
```

    </tr>
</table>

Adding these to the request body when creating the call would look like this:

```javascript
{
  "systemPrompt": "You are a helpful assistant...",
  ...
  "medium": {
    "twilio": {}
  },

>   "initiator": "agent"

Reviewer: ```suggestion
  "initiator": "INITIATOR_AGENT"
```
